src/Build/Logging/TerminalLogger/TerminalLogger.cs

```diff
@@ -823,6 +855,16 @@ private void TargetFinished(object sender, TargetFinishedEventArgs e)
                 }
             }
         }
+        else if (targetOutputs is not null
+            && e.TargetName == "InitializeSourceRootMappedPaths"
+            && _projects.TryGetValue(new ProjectContext(buildEventContext), out project)
+            && project.SourceRoot is null)
+        {
+            project.SourceRoot =
+                (targetOutputs as IEnumerable<ITaskItem>)?
+                .FirstOrDefault(root => !string.IsNullOrEmpty(root.GetMetadata("SourceControl")))
+                ?.ItemSpec.AsMemory();
+        }
```

Comment:
Here is where we consume the Target Outputs of the Source Root-generation Target to set gather more information that we'll use during Output Path rendering.